crates/sui-node/src/lib.rs

```diff
@@ -185,15 +180,15 @@ impl SuiNode {
         )
         .await;
 
-        let checkpoint_executor_handle = {
-            let executor = CheckpointExecutor::new(
-                state_sync_handle.subscribe_to_synced_checkpoints(),
-                checkpoint_store.clone(),
-                state.clone(),
-                &prometheus_registry,
-            )?;
-            tokio::spawn(executor.run())
-        };
+        let checkpoint_executor_handle = CheckpointExecutor::new(
+            state_sync_handle.subscribe_to_synced_checkpoints(),
+            checkpoint_store.clone(),
+            state.clone(),
+            &prometheus_registry,
+        )
+        .start()?;
+
+        let reconfig_channel = checkpoint_executor_handle.subscribe_to_end_of_epoch();
```

Comment:
There may be a minor race condition between calling `start()` above and subscribing here. where a node that is started right before end of epoch doesn't get the message. @bmwill at which point in `SuiNode::start()` will the node start actually receiving messages from peers?

Reply:
Chatted offline. Not likely that we'll hit this, but will fix it here anyway

Reply:
why is it unlikely ?

Reply:
Because it would take some time to connect to the peers. And other system components are started shortly after the p2p network. That said, I already have a cautious fix in the next version. This PR is probably no longer ready for a full review. Ran into some things to debug after extending tests. Will take out of draft mode once it's ready again

Reply:
This should be fixed now.